src/charm.py

```diff
@@ -159,47 +72,31 @@ def _on_publish_relation_changed(self, event):
         self._reload_config()
 
     def _on_config_changed(self, _):
-        for key in self.model.config:
-            if key not in self._stored.config:
-                value = self.model.config[key]
-                self._stored.config[key] = value
-            if self.model.config[key] != self._stored.config[key]:
-                value = self.model.config[key]
-                logger.info("Setting {} to: {}".format(key, value))
-                self._stored.config[key] = value
-
-        # this should go into install hook
-        uid = pwd.getpwnam("root").pw_uid
-        gid = grp.getgrnam("root").gr_gid
-
-        path = "/etc/nginx/ssl"
-        if not os.path.exists(path):
-            os.makedirs(path, 0o755)
-            os.chown(path, uid, gid)
-            os.chmod(path, 0o755)
-
-        config = self._stored.config
-        self._stored.config["ssl_enabled"] = (
-            "ssl_cert" in config.keys()  # noqa: W503
-            and "ssl_key" in config.keys()  # noqa: W503
-            and "ssl_cert" in self._stored.config.keys()  # noqa: W503
-            and "ssl_key" in self._stored.config.keys()  # noqa: W503
-        )
+        config = self.model.config
+        for key in config:
+            self._stored.config[key] = config[key]
+
+        ssl_cert_path = "/etc/nginx/ssl/server.crt"
+        if config.get("ssl_cert"):
+            atomic_write_root_file(ssl_cert_path, b64decode(config["ssl_cert"]), 0o644)
+        else:
+            force_remove(ssl_cert_path)
+
+        ssl_key_path = "/etc/nginx/ssl/server.key"
+        if config.get("ssl_key"):
+            atomic_write_root_file(ssl_key_path, b64decode(config["ssl_key"]), 0o640)
```

Comment:
not-blocking: I think we should use same permission for ssl_key and ssl_cert, or provide some comment why we are using two different. Potentially, consider using constant.